examples/camera-app/linux/src/clusters/camera-avsettingsuserlevel-mgmt/camera-avsettingsuserlevel-manager.cpp

```diff
@@ -56,75 +59,57 @@ void CameraAVSettingsUserLevelManager::DefaultViewportUpdated(Globals::Structs::
     this->GetServer()->UpdateMoveCapableVideoStreams(aViewport);
 }
 
-Status CameraAVSettingsUserLevelManager::MPTZSetPosition(Optional<int16_t> aPan, Optional<int16_t> aTilt, Optional<uint8_t> aZoom)
+Status CameraAVSettingsUserLevelManager::MPTZSetPosition(Optional<int16_t> aPan, Optional<int16_t> aTilt, Optional<uint8_t> aZoom,
+                                                         PhysicalPTZCallback * callback)
 {
+    mCallback = callback;
```

Comment:
The mCallback pointer is set each time a PTZ command is issued. If a new command is issued before the previous one’s timer expires (or HAL calls back), mCallback is overwritten and the callback for the previous command is lost.